### DISCUSSION TITLE
FALSE LOGICAL appear in the second round of a loop, but it was fine at the first loop

### DISCUSSION THREAD

**Discussion by P11GH** [yalmip/YALMIP#1205] [Problems using YALMIP] [answered]
Hi everyone,
I am using YALMIP to code an ADMM-like algorithm. The code has a for loop, and in every loop, there is an optimization problem with nonnegative constraints [x1>=0, x2>=0, ..., xn>=0] and some other constraints. In the first loop, the code runs fine but in the second loop, it tells that "One of the constraints evaluates to a FALSE LOGICAL variable. Your model is infeasible" at the nonnegative constraints.
This puzzles me because in the first loop, it goes through while in the second loop, it fails with such an error.
Does anyone meet a similar issue? Does anyone know what causes it?

**Comment by johanlofberg** [comment, marked as answer]
It means some of the terms x1, x2 etc aren't sdpvars but a double, and thus the constraint evaluates to a double, which then appears to be negative

**Reply by P11GH** [reply]
All x2, x2, ... xn are decision variables (vector of dimension 1 by m) defined with sdpvar. Do you know how I can obtain the desired constraints?

**Reply by johanlofberg** [reply]
Impossible, [x1>=0, x2>=0 ] etc cannot generate the mentioned error if they all are sdpvars. You are either looking at the wrong constraint, you have accidentally changed the xi variables or you have not shown what the constraint really is.

**Reply by P11GH** [reply]
Thank you very much. Let me check what you mentioned and get it updated later.

**Reply by P11GH** [reply]
what also puzzles me is that in the first loop, the code runs well without any error, but in the second loop (sometimes the third loop, depending whether I put the non-negative constraints as the last constraints or not), this error pops up.

**Reply by johanlofberg** [reply]
So to summarize the information available for debugging this: You've written some code, it does not do what you think it does.

**Reply by P11GH** [reply]
haha, that's everyone complains when he meets bugs.

**Reply by P11GH** [reply]
I got it. I accidentally made one sdpvar to double. Thank you very much.
